9.0/Dockerfile

```diff
@@ -131,6 +131,9 @@ RUN set -ex; \
     apt-get -y install acl dirmngr lsof procps wget netcat gosu tini jattach; \
     rm -rf /var/lib/apt/lists/*;
 
+# Workaround for JDK17 JIT compiler bug https://bugs.openjdk.org/browse/JDK-8285835
+ARG SOLR_OPTS="-XX:CompileCommand=exclude,com.github.benmanes.caffeine.cache.BoundedLocalCache::put"
```

Comment:
Adding [the -a parameter](https://solr.apache.org/guide/solr/latest/deployment-guide/solr-control-script-reference.html#start-parameters) to the `CMD` section works much better here (and is meant to be used with the -X options). I think it is likely used less than `SOLR_OPTS`. At least it would work with the Solr Operator, which always explicitly defines `SOLR_OPTS`.

Reply:
Can you have multiple "a" option, too?

Anyways I like you option better as it would be immune to external parameters.

Reply:
I don't think Solr supports multiple "a" options unfortunately. We can probably change that, but obviously that doesn't help with 9.0.

Maybe for future versions we can add a hidden (non-documented) envVar that will let us hot-fix the docker images without overwriting user options. (Would have helped with this bug and the log4j vulnerability)

Reply:
Changed to `-a`, which is a better option. Testet, and you can now use SOLR_OPTS in addition for user-provided opts.